lib/ex_machina/ecto_strategy.ex

```diff
@@ -74,16 +74,17 @@ defmodule ExMachina.EctoStrategy do
     assocs = get_schema_assocs(schema)
 
     Enum.reduce(assocs, struct, fn(assoc, struct) ->
-      original_value = Map.get(struct, assoc)
-      casted_value = if is_list(original_value) do
-        Enum.map(original_value, &(cast_assoc(&1, assoc, struct)))
-      else
-        cast_assoc(original_value, assoc, struct)
-      end
+      casted_value = Map.get(struct, assoc)
+                   |> cast_assoc(assoc, struct)
+
       Map.put(struct, assoc, casted_value)
     end)
   end
 
+  defp cast_assoc(original_value, assoc, struct) when is_list(original_value) do
```

Comment:
I'd probably move this back into the `case` statement below so the forking is all done in one place. Having one of these as a separate function and the others in a `case` seems sort of arbitrary.